### PR TITLE
Remove shell use in subprocess

### PR DESCRIPTION
**Pull request by Shrews**

### CODE REVIEW

**Comment by Shrews** [comment]
recheck

File: ansible_runner/runner.py [hunk 1]
```diff
@@ -204,8 +204,12 @@ def run(self):
             user = getpass.getuser()
             group = grp.getgrgid(os.getgid()).gr_name
 
-            cmd = 'cgcreate -a {user}:{group} -t {user}:{group} -g cpuacct,memory,pids:{}'.format(cgroup_path, user=user, group=group)
-            proc = Popen(cmd, stdout=PIPE, stderr=PIPE, shell=True)
+            cmd = ['cgcreate',
+                   '-a', '{user}:{group}'.format(user=user, group=group),
+                   '-t', '{user}:{group}'.format(user=user, group=group),
+                   '-g', 'cpuacct,memory,pids:{}'.format(cgroup_path),
```
Comment (samdoran) [inline ansible_runner/runner.py]: Nit:
```suggestion
                   '-a', f'{user}:{group},
                   '-t', f'{user}:{group}',
                   '-g', f'cpuacct,memory,pids:{cgroup_path}',
```

File: ansible_runner/runner.py [hunk 2]
```diff
@@ -391,8 +394,8 @@ def _delete(retries=15):
                 return True
             _delete()
         if self.resource_profiling:
-            cmd = 'cgdelete -g cpuacct,memory,pids:{}'.format(cgroup_path)
-            proc = Popen(cmd, stdout=PIPE, stderr=PIPE, shell=True)
+            cmd = ['cgdelete', '-g', 'cpuacct,memory,pids:{}'.format(cgroup_path)]
```
Comment (samdoran) [inline ansible_runner/runner.py]: Nit:
```suggestion
            cmd = ['cgdelete', '-g', f'cpuacct,memory,pids:{cgroup_path}']
```

**Review by samdoran — APPROVED** [review]

File: ansible_runner/runner.py [hunk 3]
```diff
@@ -532,8 +535,8 @@ def kill_container(self):
         container_name = self.config.container_name
         if container_name:
             container_cli = self.config.process_isolation_executable
-            cmd = '{} kill {}'.format(container_cli, container_name)
-            proc = Popen(cmd, stdout=PIPE, stderr=PIPE, shell=True)
+            cmd = ['{}'.format(container_cli), 'kill', '{}'.format(container_name)]
```
Comment (samdoran) [inline ansible_runner/runner.py]: I don't think formatting is necessary here at all.
```suggestion
            cmd = [container_cli, 'kill', container_name]
```

File: ansible_runner/runner.py [hunk 4]
```diff
@@ -204,8 +204,12 @@ def run(self):
             user = getpass.getuser()
             group = grp.getgrgid(os.getgid()).gr_name
 
-            cmd = 'cgcreate -a {user}:{group} -t {user}:{group} -g cpuacct,memory,pids:{}'.format(cgroup_path, user=user, group=group)
-            proc = Popen(cmd, stdout=PIPE, stderr=PIPE, shell=True)
+            cmd = ['cgcreate',
+                   '-a', f'{user}:{group},
```
Comment (samdoran) [inline ansible_runner/runner.py]:
```suggestion
                   '-a', f'{user}:{group}',
```

**Comment by Shrews** [comment]
recheck

**Pull request merged**